### ISSUE TITLE
Sharing database locally

### ISSUE THREAD

**Issue by elistone** [geerlingguy/drupal-vm#1471]
I'm trying to open up the VM so that other people in my office can connect to my machine to use the database, I have got this working with other Vagrant files but I am struggling to get it to work with DrupalVM.

We are getting time out errors or cannot get route, depending on settings.

DrupalVM is installed via composer & have followed the instructions [here ](http://docs.drupalvm.com/en/latest/deployment/composer-dependency/#setup-your-configuration-files) to get that working. All that I am missing is the ability to share the database. 

We are all using windows machines if that helps.

**Comment by geerlingguy** [comment]
This should be possible by setting a public IP address using this setting in your `config.yml` file: https://github.com/geerlingguy/drupal-vm/blob/master/default.config.yml#L25

Then run `vagrant reload` to restart the VM and bring up the new network interface. See: https://docs.vagrantup.com/v2/networking/public_network.html

**Comment by elistone** [comment]
Thanks for getting back to me, I have not fully managed to test this as rather busy at the moment and had to setup a work around to get it working - have the database somewhere else now.

but simply put setting a public IP address did not seem to work, as adding the same IP as `vagrant_ip` which maybe was the problem. but I did manage to allow collages to view the site locally using that IP but could not connect to the DB from there machine kept getting time out.

**Comment by geerlingguy** [comment]
@elistone - Ah, if you want _other_ people to be able to connect from other machines on the network outside of your own local machine, then that requires a bit more work:

  1. You need to make sure the `mysql_databases` database configuration allows any host (or ranges of hosts) to connect: https://github.com/geerlingguy/drupal-vm/blob/master/default.config.yml#L190-L194
  2. You need to add port `3306` to the firewall allowed TCP ports list (https://github.com/geerlingguy/drupal-vm/blob/master/default.config.yml#L240-L251) and run `vagrant provision` to update.
  3. You need to make sure MySQL allows connections from any network interface, by adding `mysql_bind_address: '0.0.0.0'` to your config file, and running `vagrant provision` to update.

And besides that, there are probably some other things that could affect MySQL-over-the-network connectivity. It's generally frowned upon to open even local development environments' databases to access anywhere, simply because it's too easy to accidentally expose data in local environments to bad actors, or to the world in some strange networking scenarios.

Since this is an unsupported (and not-recommended) use of MySQL inside Drupal VM, I'm closing this issue—but feel free to follow up with how it goes if you get a chance to try again!

Unsupported/not-recommended doesn't mean it should never be done. Just that I don't officially 'bless' this use of Drupal VM :)

**Comment by elistone** [comment]
Thanks for you response @geerlingguy I have finally managed to test this and it is working exactly we want now.

I know you don't support/recommend this but to confirm to others who might be looking to do the same thing:

I did **not** need to change any of the setting to do with  `mysql_database` but simply added port `3306`  to the firewall allowed TCP ports list, then added `mysql_bind_address: '0.0.0.0'` to my config and run `vagrant provision`

Others in my office can now connect to my machines database but use there own local files.

Thanks again.
